here-naksha-lib-handlers/src/main/java/com/here/naksha/lib/handlers/internal/IntHandlerForStorages.java

```diff
@@ -33,11 +45,55 @@ public IntHandlerForStorages(final @NotNull INaksha hub) {
   }
 
   @Override
-  protected @NotNull Result validateFeature(Storage storage) {
-    Result basicValidation = super.validateFeature(storage);
+  protected @NotNull Result validateFeature(XyzFeatureCodec codec) {
+    final EWriteOp operation = EWriteOp.get(codec.getOp());
+    if (operation.equals(EWriteOp.DELETE)) {
+      // For DELETE, only the feature ID is needed, other JSON properties are irrelevant
+      return noActiveHandlerValidation(codec);
+    }
+    // For non-DELETE write request
+    Result basicValidation = super.validateFeature(codec);
     if (basicValidation instanceof ErrorResult) {
       return basicValidation;
     }
+    final Storage storage = (Storage) codec.getFeature();
     return pluginValidation(storage);
   }
+
+  private Result noActiveHandlerValidation(XyzFeatureCodec codec) {
+    // Search for active event handlers still using this storage
+    String storageId = codec.getId();
+    if (storageId == null) {
+      if (codec.getFeature() == null) {
+        return new ErrorResult(XyzError.ILLEGAL_ARGUMENT, "No storage ID supplied.");
+      }
+      storageId = codec.getFeature().getId();
+    }
+    // "properties.storageId" = <storage-id-to-be-deleted>
+    final PRef pRef =
+        RequestHelper.pRefFromPropPath(new String[] {XyzFeature.PROPERTIES, EventHandlerProperties.STORAGE_ID});
+    final POp activeHandlersPOp = POp.eq(pRef, storageId);
+    final ReadFeatures readActiveHandlersRequest =
+        new ReadFeatures(EVENT_HANDLERS).withPropertyOp(activeHandlersPOp);
+    try (final IReadSession readSession =
+        nakshaHub().getAdminStorage().newReadSession(NakshaContext.currentContext(), false)) {
+      final Result readResult = readSession.execute(readActiveHandlersRequest);
+      if (!(readResult instanceof SuccessResult)) {
+        return readResult;
+      }
+      final List<EventHandler> eventHandlers;
+      try {
+        eventHandlers = readFeaturesFromResult(readResult, EventHandler.class);
+      } catch (NoCursor | NoSuchElementException emptyException) {
+        // No active handler using the storage, proceed with deleting the storage
+        return new SuccessResult();
+      }
+      final List<String> handlerIds = new ArrayList<>();
+      for (EventHandler handler : eventHandlers) {
+        handlerIds.add(handler.getId());
+      }
```

Comment:
consdier using Stream + map + toList for `handlerIds` population instead of forEach

Reply:
roger that :)